Zend/zend_compile.c

```diff
@@ -3221,7 +3336,11 @@ uint32_t zend_compile_args(zend_ast *ast, zend_function *fbc) /* {{{ */
 				}
 			}
 		} else if (zend_is_variable(arg)) {
-			if (fbc) {
+			if (zend_ast_is_short_circuited(arg)) {
+				// Must use SEND_VAL_EX to make sure the value isn't sent by ref
+				zend_compile_expr(&arg_node, arg);
+				opcode = ZEND_SEND_VAL_EX;
```

Comment:
We can still use ZEND_SEND_VAL if we have fbc and know the arg is by-val.

Reply:
I think what you want to do here instead is just change the condition to `zend_is_variable(arg) && !zend_ast_is_short_circuited(arg)`, so it is treated like a normal expression here.